Mathlib/Analysis/Calculus/Monotone.lean

```diff
@@ -191,19 +191,22 @@ theorem Monotone.ae_hasDerivAt {f : ℝ → ℝ} (hf : Monotone f) :
       · have : Ioo (x - 1) x ∈ 𝓝[<] x := by
           apply Ioo_mem_nhdsWithin_Iio; exact ⟨by linarith, le_refl _⟩
         filter_upwards [this]
-        rintro y ⟨hy : x - 1 < y, h'y : y < x⟩
+        rintro y hy
+        rw [mem_Ioo] at hy
         rw [mem_Iio]
         norm_num; nlinarith
     -- apply the sandwiching argument, with `g` and the helper function
     apply tendsto_of_tendsto_of_tendsto_of_le_of_le' hx.1 this
     · filter_upwards [self_mem_nhdsWithin]
-      rintro y (hy : y < x)
-      apply div_le_div_of_nonpos_of_le (sub_neg.2 hy).le
+      rintro y hy
+      rw [mem_Iio, ← sub_neg] at hy
```

Comment:
Personally I find the original cleaner. Do you have a reason to prefer the explicit `rw`?

Reply:
Mostly just that it relies on the exact definition of the `Ixx`es. `rw` should be robust to changes there (though I suppose those are unlikely), but might also give a better error message if `filter_upwards` ends up creating a different goal here (which may be equally unlikely). Happy to revert these parts if you'd like